csclient/csclient.go

```diff
@@ -531,3 +548,25 @@ func (cs *Client) Log(typ params.LogType, level params.LogLevel, message string,
 	resp.Body.Close()
 	return nil
 }
+
+// Login explicitly obtains authorization credentials
+// for the charm store and stores them in the client's
+// cookie jar.
+func (cs *Client) Login() error {
+	var m macaroon.Macaroon
+	if err := cs.Get("/macaroon", &m); err != nil {
+		return errgo.Notef(err, "cannot retrieve the authentication macaroon")
+	}
+	ms, err := httpbakery.DischargeAll(&m, cs.params.HTTPClient, cs.params.VisitWebPage)
+	if err != nil {
+		return errgo.Mask(err)
```

Comment:
This code has just moved from charmcmd without any changes, but I agree. I'll add a Notef.